de.dlr.sc.virsat.fdir.core/src/de/dlr/sc/virsat/fdir/core/matrix/MatrixIterator.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package de.dlr.sc.virsat.fdir.core.matrix;
+
+/**
+ * @author piet_ci
+ * 
+ * Abstract matrix iterator class   
+ */
+public abstract class MatrixIterator {

```

Comment:
Isnt it a MatrixExponentialIterator?

Reply:
MatrixIterator is abstract and meant to be extended by a concrete implementation that could be MatrixExponentialIterator in fact. We have such a class already on the jblas branch named jblasMatrixIterator which is basically exactly that. We can change the naming later on. For now its named jblasMatrixIterator to stress the difference in the underlying matrix handling and to be a hint for external lib usage. Presumably this will get resolved in the progess.